server/swap/swap.go

```diff
@@ -1030,44 +1030,60 @@ func (s *Swapper) TxMonitored(user account.AccountID, asset uint32, txid string)
 	checkMatch := func(match *matchTracker) bool {
 		// Both maker and taker statuses must be locked to check redeem status
 		// and swap tx id.
-		match.makerStatus.mtx.RLock()
-		defer match.makerStatus.mtx.RUnlock()
-		match.takerStatus.mtx.RLock()
-		defer match.takerStatus.mtx.RUnlock()
+		makerStatus, takerStatus := match.makerStatus, match.takerStatus
+		makerStatus.mtx.RLock()
+		defer makerStatus.mtx.RUnlock()
+		takerStatus.mtx.RLock()
+		defer takerStatus.mtx.RUnlock()
 
 		// The swap contract of either the maker or taker must correspond to
 		// specified asset to be of interest.
 		switch asset {
-		case match.makerStatus.swapAsset:
-			// Maker's swap transaction is the asset of interest.
+		case makerStatus.swapAsset:
+			// This is the asset of the maker's swap / takers redeem.
+
+			// For maker, check the swap txid.
 			if user == match.Maker.User() {
 				// The swap contract tx is considered monitored until the swap
 				// is complete, regardless of confirms.
-				return match.makerStatus.swap != nil && match.makerStatus.swap.TxID() == txid
+				return makerStatus.swap != nil && makerStatus.swap.TxID() == txid
 			}
 
-			// Taker's redemption transaction is the asset of interest.
-			_, takerRedeemDone := s.redeemStatus(match.makerStatus, match.takerStatus)
-			if !takerRedeemDone && user == match.Taker.User() &&
-				match.takerStatus.redemption.TxID() == txid {
-				return true
+			// For taker, check the redeem txid, but only if it is not complete.
+			_, takerRedeemDone := s.redeemStatus(makerStatus, takerStatus)
+			if takerRedeemDone {
+				// Taker has a redemption that has reached required confirms.
+				return false // so no longer monitored?
```

Comment:
I honestly don't follow the reasoning for considering it **not monitored** if we know about the redemption and it has reached the required confirms, but considering it **monitored** if it has not reached the required confirms. @buck54321 Thoughts about why we don't just consider a tx monitored if it corresponds to a redemption regardless of the confirmations?  After all, if it is found in the `Swapper.matches` map, the swap is still active.

Reply:
I think you've got it right, but I'll note that the `matches` map is being pruned in `processBlocks`, so `TxMonitored` won't hit after `MatchComplete` anyway. We either need the full DB solution, or `Swapper` should hang onto matches until `FundConf` on the redemptions.

Reply:
> but I'll note that the `matches` map is being pruned in `processBlocks`, so `TxMonitored` won't hit after `MatchComplete` anyway

That's what I was getting at with "if it is found in the Swapper.matches map, the swap is still active".  In which case the user will just have to wait for FundConf.

A DB powered solution has been on the table for a while, although I'm super hesitant to load the DB for with a check of all matches for each order placed.

Anyway, on this current implementation, I think I'll remove the redeemStatus/makerRedeemStatus functions since I think we agree that it makes no sense to consider the tx not monitored because the redeem has reached `SwapConf`. Correct?